src/rust/engine/process_execution/src/local.rs

```diff
@@ -266,14 +268,24 @@ impl super::CommandRunner for CommandRunner {
     let argv = req.argv;
     let req_description = req.description;
     let maybe_jdk_home = req.jdk_home;
+    let store2 = self.store.clone();
+    let scratch = req.local_scratch_files;
+    let workunit_store2 = workunit_store.clone();
+
     self
       .store
       .materialize_directory(workdir_path.clone(), req.input_files, workunit_store)
+      .and_then(move |_m2| store2.materialize_directory(workdir_path5, scratch, workunit_store2))
       .and_then(move |_metadata| {
         maybe_jdk_home.map_or(Ok(()), |jdk_home| {
           symlink(jdk_home, workdir_path3.clone().join(".jdk"))
             .map_err(|err| format!("Error making symlink for local execution: {:?}", err))
         })?;
+        debug!("============ after ============");
+        for file in walkdir::WalkDir::new(workdir_path4.clone()) {
```

Comment:
I think that like our jdk_home hack, this could probably be a path that is exposed to local runs via a symlink or something? Remote runs would then attempt to read-from an empty path, and would write-to a path that would be discarded afterward. And local runs would read/write to the symlink destination. You could still call it something like ".scratch" or something.

My feeling is that digesting and re-digesting like this is likely to impose a cost to capture files that we don't place much faith in anyway.